jupyterhub/templates/singleuser/image-credentials-secret.yaml

```diff
@@ -8,5 +8,5 @@ metadata:
     {{- include "jupyterhub.labels" $_ | nindent 4 }}
 type: kubernetes.io/dockerconfigjson
 data:
-  .dockerconfigjson: {{ .Values.singleuser.image.pullSecret.dockerConfigJson }}
+  .dockerconfigjson: {{ .Values.singleuser.image.pullSecret.dockerConfigJson| b64enc }}
```

Comment:
This assumes we have something like this:

```yaml
singleuser:
  image:
    name: ...
    tag: ...
    pullSecrets:
      dockerConfigJson: ...
```

Do you mind we do it like this instead?

```
singleuser:
  image:
    name: someimagename
    tag: sometag
    credentials:
      registry: quay.io
      username: someone
      password: sillyness
```

You could copy paste the template from [the Helm tips and tricks section](https://github.com/helm/helm/blob/master/docs/charts_tips_and_tricks.md#creating-image-pull-secrets) and put it in [_helpers.tpl](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/master/jupyterhub/templates/_helpers.tpl)

Reply:
crap yes

Reply:
that comment got lost in me pushing stuff